launcher/launcher.go

```diff
@@ -193,11 +208,34 @@ func (l *launcher) stopBeater() {
 	l.log.Infof("Launcher shut %s down gracefully", l.name)
 }
 
-// waitForUpdates is the function that keeps Launcher runLoop busy
+// Returns an error indicating if the beater was stopped gracefully or not
+func (l *launcher) stopBeaterWithTimeout(duration time.Duration) error {
```

Comment:
Why use an error here?

Reply:
used to differentiate between returning with/without waiting for the launcher's wait group. could use any pair values but an error seemed most explicit way in go.